### ISSUE TITLE
Data issue?

### ISSUE THREAD

**Issue by threegold116** [RUCAIBox/CIR#5]
The content of train.parquet is same as the train.parquet file in ToRL github. What is the training data for the cir model?

**Comment by Enoch202** [comment]
Thank you for your interest! Our train.parquet contains 88,054 samples, which represents the full dataset. The results reported in the paper are based on a subset: we set the random seed to 80, shuffle the full training set, and use the first 38,000 samples for training. We found that this configuration yields the best performance.

**Comment by Enoch202** [comment]
> > Thank you for your interest! Our train.parquet contains 88,054 samples, which represents the full dataset. The results reported in the paper are based on a subset: we set the random seed to 80, shuffle the full training set, and use the first 38,000 samples for training. We found that this configuration yields the best performance.感谢您的关注！我们的 train.parquet 包含 88,054 个样本，代表了完整的数据集。论文中报告的结果基于一个子集：我们将随机种子设置为 80，打乱完整的训练集，并使用前 38,000 个样本进行训练。我们发现这种配置产生了最佳性能。
> 
> Thanks for your reply. Does the sampling process occur during training or before training? And how to set the sampling process?

It occurs before training and can be set in the run.sh script. https://github.com/RUCAIBox/CIR/blob/5c642ad0a225fee3a3e9afa8c4d0df59a025bce5/verl/verl-main/scripts/run/run.sh#L58

**Comment by threegold116** [comment]
> > > Thank you for your interest! Our train.parquet contains 88,054 samples, which represents the full dataset. The results reported in the paper are based on a subset: we set the random seed to 80, shuffle the full training set, and use the first 38,000 samples for training. We found that this configuration yields the best performance.感谢您的关注！我们的 train.parquet 包含 88,054 个样本，代表了完整的数据集。论文中报告的结果基于一个子集：我们将随机种子设置为 80，打乱完整的训练集，并使用前 38,000 个样本进行训练。我们发现这种配置产生了最佳性能。感谢您的关注！我们的 train.parquet 包含 88,054 个样本，代表了完整的数据集。论文中报告的结果基于一个子集：我们将随机种子设置为 80，打乱完整的训练集，并使用前 38,000 个样本进行训练。我们发现这种配置产生了最佳性能。感谢您的关注！我们的 train.parquet 包含 88,054 个样本，代表了完整的数据集。论文中报告的结果基于一个子集：我们将随机种子设置为 80，打乱完整的训练集，并使用前 38,000 个样本进行训练。我们发现这种配置产生了最佳性能。
> > 
> > 
> > Thanks for your reply. Does the sampling process occur during training or before training? And how to set the sampling process?
> 
> It occurs before training and can be set in the run.sh script.
> 
> [CIR/verl/verl-main/scripts/run/run.sh](https://github.com/RUCAIBox/CIR/blob/5c642ad0a225fee3a3e9afa8c4d0df59a025bce5/verl/verl-main/scripts/run/run.sh#L58)
> 
> Line 58 in [5c642ad](/RUCAIBox/CIR/commit/5c642ad0a225fee3a3e9afa8c4d0df59a025bce5)
> 
>  +data.seed=80 \

Thanks. I would also like to know if the final model is the one obtained by 30 epoch training?

**Comment by Enoch202** [comment]
> > > > > > Thank you for your interest! Our train.parquet contains 88,054 samples, which represents the full dataset. The results reported in the paper are based on a subset: we set the random seed to 80, shuffle the full training set, and use the first 38,000 samples for training. We found that this configuration yields the best performance.感谢您的关注！我们的 train.parquet 包含 88,054 个样本，代表了完整的数据集。论文中报告的结果基于一个子集：我们将随机种子设置为 80，打乱完整的训练集，并使用前 38,000 个样本进行训练。我们发现这种配置产生了最佳性能。感谢您的关注！我们的 train.parquet 包含 88,054 个样本，代表了完整的数据集。论文中报告的结果基于一个子集：我们将随机种子设置为 80，打乱完整的训练集，并使用前 38,000 个样本进行训练。我们发现这种配置产生了最佳性能。感谢您的关注！我们的 train.parquet 包含 88,054 个样本，代表了完整的数据集。论文中报告的结果基于一个子集：我们将随机种子设置为 80，打乱完整的训练集，并使用前 38,000 个样本进行训练。我们发现这种配置产生了最佳性能。
> > > > > 
> > > > > 
> > > > > Thanks for your reply. Does the sampling process occur during training or before training? And how to set the sampling process?
> > > > 
> > > > 
> > > > It occurs before training and can be set in the run.sh script.
> > > > [CIR/verl/verl-main/scripts/run/run.sh](https://github.com/RUCAIBox/CIR/blob/5c642ad0a225fee3a3e9afa8c4d0df59a025bce5/verl/verl-main/scripts/run/run.sh#L58)
> > > > Line 58 in [5c642ad](/RUCAIBox/CIR/commit/5c642ad0a225fee3a3e9afa8c4d0df59a025bce5)
> > > > +data.seed=80 \
> > > 
> > > 
> > > Thanks. I would also like to know if the final model is the one obtained by 30 epoch training?
> > 
> > 
> > Actually not, the 30 epochs shown do not mean that we truly trained for 30 epochs. Due to the enormous computational cost of RL training, we stopped training after approximately 1 epoch.实际上并非如此，显示的 30 个纪元并不意味着我们真正训练了 30 个纪元。由于 RL 训练的计算成本巨大，我们在大约 1 个 epoch 后停止了训练。
> 
> So the max_call_time reduction is also done in one epoch? And this is done by resume training?


Yes. We gradually increase `max_call_time`, and in the current code this is implemented through resume training. However, we are also exploring more adaptive scaling approaches.

**Comment by Enoch202** [comment]
> > Yes. We gradually increase `max_call_time`, and in the current code this is implemented through resume training. However, we are also exploring more adaptive scaling approaches.
> 
> Oh, I see. Thank you for answering my question carefully.

You're welcome.
